libyul/AsmAnalysis.cpp

```diff
@@ -177,11 +183,9 @@ void AsmAnalyzer::operator()(ExpressionStatement const& _statement)
 		m_errorReporter.typeError(
 			3083_error,
 			_statement.location,
-			"Top-level expressions are not supposed to return values (this expression returns " +
-			to_string(types.size()) +
-			" value" +
-			(types.size() == 1 ? "" : "s") +
-			"). Use ``pop()`` or assign them."
+			"Top-level expressions are not supposed to return values (this expression returns {} value{}). Use ``pop()`` or assign them.",
+			types.size(),
+			(types.size() == 1 ? "" : "s")
```

Comment:
I still very much dislike the fact that values are given out-of-place, but what I would expect from a formatter is the ability to handle plural well.

Reply:
What about defining a macro
`FORMAT("Top-level expressions are not supposed to return values (this expression returns ${type.size()} value{$c{type.size()}}). Use ``pop()`` or assign them.")`

Reply:
While fmtlib might be part of the standard, we are really using a tiny fraction of it. I would say that `to_string` always works for us - we don't need `%010d` or similar. When we need something like that, it is probably better to use explicitly named functions you can "ctrl-click" on instead of having to lookup the obscure formatting abbreviation in the documentation.